src/features/vote/hooks/useRewardsRetrievedEvents.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from "react-query";
+import { ethers } from "ethers";
+import {
+  queryRewardsRetrieved,
+  RewardsRetrieved,
+} from "web3/queryVotingContractEvents";
+
+export default function useRewardsRetrievedEvents(
+  contract: ethers.Contract | null,
+  address: string | null
+) {
+  const { data, error, isFetching } = useQuery<RewardsRetrieved[]>(
+    "rewardsRetrievedEvents",
+    () => {
+      return queryRewardsRetrieved(contract, address).then((res) => {
+        if (res) {
+          return res;
+        } else {
+          return [];
+        }
+      });
+    },
+    { enabled: contract !== null ? true : false }
```

Comment:
I think you can get rid of the ternary here and just use `contract !=null`

Reply:
Good call.